### PR TITLE
chore(JS): Add confirmSignIn() flows to JS sign in docs

### PR DESCRIPTION
**Pull request by cwomack**
#### Related GitHub issue #, if available:

Addresses JS and RN docs for #5468

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/lib/auth/js/emailpassword.mdx [hunk 1]
```diff
@@ -141,6 +141,107 @@ async function signIn() {
...
+      user.challengeName === 'SOFTWARE_TOKEN_MFA'
+    ) {
+      // You need to get the code from the UI inputs
+      // and then trigger the following function with a button click.
+      const code = getCodeFromUserInput();
+      // If MFA is enabled, sign-in should be confirmed with the confirmation code
+      const loggedUser = await Auth.confirmSignIn(
+        user, // Return object from Auth.signIn()
+        code, // Confirmation code
+        mfaType // MFA Type e.g. SMS_MFA, SOFTWARE_TOKEN_MFA
```
Comment (israx) [inline src/fragments/lib/auth/js/emailpassword.mdx:171]: probably we can define this argument in a variable

File: src/fragments/lib/auth/js/emailpassword.mdx [hunk 2]
```diff
@@ -141,6 +141,107 @@ async function signIn() {
...
+      console.log(user);
+    }
+  } catch(err => console.log(err));
+}
+```
+
+
+### Confirm sign in with new password
+
+New users can be created from the [Cognito console](https://console.aws.amazon.com/cognito/home) for your app, requiring the user to input a new password and required attributes (optional) when signing in. If the user object returned by `Auth.signIn()` contains the `NEW_PASSWORD_REQUIRED` for the `challengeParam`, you can submit the data using the `Auth.completeNewPassword()` method as well.
```
[the remark this answers is not in the document]
Comment (israx) [reply src/fragments/lib/auth/js/emailpassword.mdx:209]: > If the user object returned by `Auth.signIn()` contains the `NEW_PASSWORD_REQUIRED` for the `challengeParam`

challengeParam -> challengeName

**Review by israx — COMMENTED** [review]
LGTM, just some nit comments. I see so far we only have addressed 3 possible `challengeName` we have left, `SELECT_MFA_TYPE`, `SOFTWARE_TOKEN_MFA` and `MFA_SETUP`. Do we want to also add documentation for those?

File: src/fragments/lib/auth/js/emailpassword.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (cwomack) [reply src/fragments/lib/auth/js/emailpassword.mdx:209]: Good point on the wording here.  Updated it with a new commit.  Also great catch on the `challengeName`!

File: src/fragments/lib/auth/js/emailpassword.mdx (hunk 1 above)
Comment (cwomack) [reply src/fragments/lib/auth/js/emailpassword.mdx]: Updated with new commit.

**Comment by cwomack** [comment]
> we only have addressed 3 possible challengeName we have left, SELECT_MFA_TYPE, SOFTWARE_TOKEN_MFA and MFA_SETUP.

Great callout @israx.  I didn't include these because there's a link to the [Advanced Use Cases ](https://docs.amplify.aws/lib/auth/mfa/q/platform/js/#advanced-use-cases) under the MFA section of the docs that details these more.  Wasn't sure if we should duplicate the docs in the Sign In section, or just refer to those via the link.

**Review by israx — APPROVED** [review]
lgtm

**Review by abdallahshaban557 — APPROVED** [review]
LGTM

**Review by katiegoines — APPROVED** [review]

**Pull request merged**